app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/survey/SurveyDetailViewModel.kt

```diff
@@ -65,6 +69,32 @@ class SurveyDetailViewModel @Inject constructor(
         }
     }
 
+    fun submitSurvey(surveyId: String) {
+        if (_currentPage.value == _surveyQuestions.value.size) {
```

Comment:
This check is weird to me, surely this method should only be called when the submit button is visible? 🤔

Reply:
@ryan-conway Yes, I thought so too. But I click on the submit button very quickly when I was testing and I got the error. I've applied the submit button with `Crossfade` previously and while it was fading I could click on the submit button. In fact `submit button` and `start survey` button is the same. I am switching the UI according to the current selected page. So when they switch, I could click on the `submit button` for half a second and if I click during that time, I got the error. That's why I added this 😬 .